app/services/agency_identity_linker.rb

```diff
@@ -26,11 +26,13 @@ def self.for(user:, service_provider:, skip_create:)
     )
 
     uuid = spi&.uuid || SecureRandom.uuid
-    AgencyIdentity.create(
+    AgencyIdentity.find_or_create_by(
       agency_id: service_provider.agency_id,
       user_id: user.id,
       uuid: uuid,
-    )
+    ) do |ai|
+      ai.uuid = uuid
```

Comment:
Does this set the UUID to the same UUID it would have otherwise had? If so, can it be deleted?